### PR TITLE
Show count of not_cloned/cloned/cloning/failed in admin UI

### PR DESCRIPTION
**Pull request by mrnugget**
This adds repository counts to the site-admin repositories page.

Under the hood, the SQL query used to count the repositories is not as efficient as it can be, because the first attempt in https://github.com/sourcegraph/sourcegraph/pull/39660 was a deadend. But I have an idea for a better solution.

Until I have that efficient solution done, though, I don't think it's harmful to ship this version, which does a single `SELECT` when viewing the page.

This also uses the new UI components that @erzhtor and @thenamankumar added for the analytics pages (cc @rob) to display the counts.

## Demo video

https://user-images.githubusercontent.com/1185253/185087781-4b223777-49ed-4b1b-aa51-cff459ac8d6f.mp4

## Test plan

- Unit tests
- Manual testing

### CODE REVIEW

File: cmd/frontend/graphqlbackend/repository_stats.go [hunk 1]
```diff
@@ -32,8 +44,18 @@ func (r *schemaResolver) RepositoryStats(ctx context.Context) (*repositoryStatsR
 		return nil, err
 	}
 
+	statisticsCounts, err := r.db.Repos().StatisticsCounts(ctx)
+	if err != nil {
+		return nil, err
+	}
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/repository_stats.go:50]: this means fetching the heavy counts for requests that are only interested in gitDirBytes, like the already slow /site-admin page iirc. 

Should we move this into a `computeRepoStatistics()` function on the stats resolver itself and memoize it in a sync.Once gate? Maybe the same applies to the calculation of the existing two fields, now that this serves two purposes.

File: client/web/src/site-admin/SiteAdminRepositoriesPage.tsx [hunk 2]
```diff
@@ -158,6 +221,13 @@ export const SiteAdminRepositoriesPage: React.FunctionComponent<React.PropsWithC
                 .
             </Text>
             <Container className="mb-3">
+                {error && !loading && (
+                    <Alert variant="warning" as="p">
+                        {error.message}
+                    </Alert>
+                )}
+                {loading && !error && <LoadingSpinner />}
```
Comment (eseliger) [inline client/web/src/site-admin/SiteAdminRepositoriesPage.tsx:229]: does this work with polling? Wouldn't it flash a loader each time it polls?
Maybe this helps, if so:
```suggestion
                {loading && !error && !legends && <LoadingSpinner />}
```

**Review by eseliger — APPROVED** [review]

File: client/web/src/site-admin/SiteAdminRepositoriesPage.tsx (hunk 2 above)
Comment (mrnugget) [reply client/web/src/site-admin/SiteAdminRepositoriesPage.tsx:229]: It doesn't. See demo video. But now I'll double-check.

File: cmd/frontend/graphqlbackend/repository_stats.go (hunk 1 above)
Comment (mrnugget) [reply cmd/frontend/graphqlbackend/repository_stats.go:50]: Alright!

File: client/web/src/site-admin/SiteAdminRepositoriesPage.tsx (hunk 2 above)
Comment (mrnugget) [reply client/web/src/site-admin/SiteAdminRepositoriesPage.tsx:229]: Nope. It works. Doesn't flash. Shows loading spinner on initial load.

**Review by erzhtor — APPROVED** [review]
Looks nice!

**Comment by mrnugget** [comment]
@sourcegraph/repo-management I'm going to go ahead and merge this, happy to address any feedback post-merge.

**Review by ryanslade — COMMENTED** [review]
Nice!

**Pull request merged**